commands/command_smudge.go

```diff
@@ -63,6 +63,15 @@ func delayedSmudge(s *git.FilterProcessScanner, to io.Writer, from io.Reader, q
 			q.Add(filename, path, ptr.Oid, ptr.Size)
 			return 0, true, ptr, nil
 		}
+
+		// Write 'statusFromErr(nil)', even though 'perr != nil', since
+		// we are about to write non-delayed smudged contents to "to".
```

Comment:
Ah, good question -- I think the comment is out of date but the code is correct here: this case occurs when we have the contents locally and can smudge without delay. It is certainly possible that this could affect cases where `lfs.concurrenttransfers=1`, but I think this would be negligible. I fixed this up in 4cefa24.